### PR TITLE
 fix the check in marketplace at `createExternalOrder` method.

### PR DESCRIPTION
**Pull request by ravitb**
Added a check for completed or pending order to prevent a reuse in the
 order.

* Main purpose:
 
* Technical description:

* Dilemmas you faced with?

* Tests

* Documentation (Source/readme.md)

### CODE REVIEW

File: scripts/src/models/orders.ts [hunk 1]
```diff
@@ -133,6 +133,18 @@ export class Order extends CreationDateModel {
...
+	public static getCompletedOrders<T extends Order>(userId: string, offerId: string): Promise<Order[]> {
+		const query = Order.createQueryBuilder()
+			.where(new Brackets(qb => {
+				qb.where("status = :completed", { completed: "completed" })
+					.orWhere("status = :pending", { pending: "pending" });
+			}))
+			.andWhere("offer_id = :offerId", { offerId })
+			.andWhere("user_id = :userId", { userId })
+			.limit(1);
+		return query.getMany() as Promise<Order[]>;
```
Comment (Ronserruya) [inline scripts/src/models/orders.ts]: why limit 1 and get many? instead of getOne?

File: scripts/src/models/orders.ts [hunk 2]
```diff
@@ -133,6 +133,17 @@ export class Order extends CreationDateModel {
...
 
+	public static getCompletedOrder<T extends Order>(userId: string, offerId: string): Promise<Order> {
+		const query = Order.createQueryBuilder()
+			.where(new Brackets(qb => {
+				qb.where("status = :completed", { completed: "completed" })
+					.orWhere("status = :pending", { pending: "pending" });
+			}))
+			.andWhere("offer_id = :offerId", { offerId })
+			.andWhere("user_id = :userId", { userId });
+		return query.getOne() as Promise<Order>;
```
Comment (nitzantomer) [inline scripts/src/models/orders.ts]: do you really need the type assertion here? it should know that it's `Promise<Order>`.
Also, why don't you just make it an async function?

File: scripts/src/public/services/orders.ts [hunk 3]
```diff
@@ -162,9 +162,10 @@ export async function createExternalOrder(jwt: string, user: User, logger: Logge
 	const payload = await validateExternalOrderJWT(jwt, user.appUserId, logger);
 
 	let order = await db.Order.findOne({ userId: user.id, offerId: payload.offer.id });
+	const completed_orders = await db.Order.getCompletedOrder( user.id, payload.offer.id);
```
Comment (nitzantomer) [inline scripts/src/public/services/orders.ts]: the convention so far has been to use camel case.
it's your repo, but if you don't stay consistent you'll end up having horrible code.
remember, this isn't python.

File: scripts/src/public/services/orders.ts [hunk 4]
```diff
@@ -162,9 +162,10 @@ export async function createExternalOrder(jwt: string, user: User, logger: Logge
...
 
 	if (!order || order.status !== "opened") {
-		if (order && (order.status === "completed" || order.status === "pending")) {
+		if (order && (completed_orders != null)) {
```
Comment (nitzantomer) [inline scripts/src/public/services/orders.ts]: you can remove the extra parentheses

File: scripts/src/public/services/orders.ts (hunk 3 above)
Comment (yosriz) [inline scripts/src/public/services/orders.ts]: you can move the completed orders query after the first if, as it's relevant only after that and not in all cases

**Comment by doodyparizada** [comment]
@ravitb by the way, your gitconfig probably has a wrong email, so github doesn't resolve your user: 
image

**Pull request merged**